File: MXApp/src/main/java/info/mx/tracks/common/On3StateClickListener.java
package info.mx.tracks.common;

import android.view.View;
import android.view.View.OnClickListener;
import android.widget.ImageView;

public class On3StateClickListener implements OnClickListener {

    protected int level;

    @Override
    public void onClick(View view) {
        level = ((ImageView) view).getDrawable().getLevel();
        level++;
        if (level == 3) {
            level = 0;
        }
        ((ImageView) view).setImageLevel(level);
    }

}

File: app/src/main/java/info/mx/tracks/common/On3StateClickListener.kt
package info.mx.tracks.common

import android.view.View
import android.widget.ImageView

open class On3StateClickListener : View.OnClickListener {
    protected var level = 0
    override fun onClick(view: View) {
        level = (view as ImageView).drawable.level
        level++
        if (level == 3) {
            level = 0
        }
        view.setImageLevel(level)
    }
}
